Trellix/connectors/trellix_edr_connector.py

```diff
@@ -0,0 +1,260 @@
+        for alert in alerts:
+            alert_detection_date = isoparse(alert.attributes.detectionDate)
+            if alert_detection_date > self.last_event_date("alerts"):
+                with self.context as cache:
+                    cache["alerts"] = alert_detection_date.isoformat()
+
+    async def get_threat_detections(self, threat_id: str, start_date: datetime, end_date: datetime) -> list[str]:
+        """
+        Get threat detections.
+
+        Args:
+            threat_id: str
+            start_date: datetime
+            end_date: datetime
+
+        Returns:
+            list[str]
+        """
+        result = []
+
+        offset = 0
+        while True:
+            detections = await self.trellix_client.get_edr_threat_detections(
+                threat_id,
+                start_date,
+                end_date,
+                self.module.configuration.records_per_request,
+                offset,
+            )
+
+            result_data = [
+                orjson.dumps({**event.dict(), "threatId": threat_id}).decode("utf-8") for event in detections
+            ]
+
+            result.extend(await self.push_data_to_intakes(result_data))
+            offset = offset + self.module.configuration.records_per_request
+
+            if len(detections) == 0:
+                break
+
+        return result
+
+    async def get_threat_affectedhosts(self, threat_id: str, start_date: datetime, end_date: datetime) -> list[str]:
+        """
+        Get threat affectedhosts.
+
+        Args:
+            threat_id: str
+            start_date: datetime
+            end_date: datetime
+
+        Returns:
+            list[str]
+        """
+        result = []
+
+        offset = 0
+        while True:
+            affectedhosts = await self.trellix_client.get_edr_threat_affectedhosts(
+                threat_id,
+                start_date,
+                end_date,
+                self.module.configuration.records_per_request,
+                offset,
+            )
+
+            result_data = [
+                orjson.dumps({**event.dict(), "threatId": threat_id}).decode("utf-8") for event in affectedhosts
+            ]
+
+            result.extend(await self.push_data_to_intakes(result_data))
+            offset = offset + self.module.configuration.records_per_request
+
+            if len(affectedhosts) == 0:
+                break
+
+        return result
+
+    def run(self) -> None:  # pragma: no cover
+        """Runs TrellixEdr."""
+        previous_processing_end = None
+
+        while self.running:
+            try:
+                loop = asyncio.get_event_loop()
+
+                while self.running:
+                    processing_start = time.time()
+                    if previous_processing_end is not None:
+                        EVENTS_LAG.labels(intake_key=self.configuration.intake_key).observe(
+                            processing_start - previous_processing_end
+                        )
```

Comment:
I think that moving EVENTS_LAG into `get_trellix_edr_events` and use `datetime.now(timezone.utc) - threats_stop_date` to compute the lag.